lang/java/avro/src/test/java/org/apache/avro/generic/TestGenericData.java

```diff
@@ -303,13 +304,24 @@ public void testToStringIsJson() throws JsonParseException, IOException {
     Field stringField = new Field("string", Schema.create(Type.STRING), null, null);
     Field enumField = new Field("enum", Schema.createEnum("my_enum", "doc", null, Arrays.asList("a", "b", "c")), null,
         null);
+
+    Schema timeSchema = Schema.create(Type.INT);
+    timeSchema.addProp("logicalType", "time-millis");
+    Field timeField = new Field("time", timeSchema, null, null);
+
+    Schema dateSchema = Schema.create(Type.LONG);
+    dateSchema.addProp("logicalType", "date");
+    Field dateField = new Field("date", dateSchema, null, null);
+
     Schema schema = Schema.createRecord("my_record", "doc", "mytest", false);
-    schema.setFields(Arrays.asList(stringField, enumField));
+    schema.setFields(Arrays.asList(stringField, enumField, timeField, dateField));
 
     GenericRecord r = new GenericData.Record(schema);
     // \u2013 is EN DASH
     r.put(stringField.name(), "hello\nthere\"\tyou\u2013}");
     r.put(enumField.name(), new GenericData.EnumSymbol(enumField.schema(), "a"));
+    r.put(timeField.name(), Instant.now());
```

Comment:
I'm not sure this is actually validating something useful...  My reasoning is that `r` is not a valid `GenericRecord` at this point because it would expect integers in these fields.

```
// Would be false
GenericData.get().validate(r.getSchema(), r);
```

I'm trying to decide at what point we have to be responsible for ensuring that toString is valid JSON even when the datum is bad -- we could put any Object that is not handled by your change and get badly delimited JSON.  What do you think?  Would a test case on a specific record make more sense?